### PR TITLE
Add CommonJS and AMD support

### PR DESCRIPTION
**Pull request by Rowno**

### CODE REVIEW

**Comment by Rowno** [comment]
I signed it!

**Comment by elektronik2k5** [comment]
+1 for this and thank you @Rowno!
This will allow importing the polyfill using the standard ES6 modules syntax or CommonJS/AMD.
```javascript
// so instead of this mess (in webpack):
const dialogPolyfill = require('exports?dialogPolyfill!dialog-polyfill')
// we could have a normal
import dialogPolyfill from 'dialogPolyfill'
// or even
import { registerDialog, } from 'dialogPolyfill'
```

File: dialog-polyfill.js [hunk 1]
```diff
@@ -483,4 +483,12 @@
   window['dialogPolyfill'] = dialogPolyfill;
   dialogPolyfill['forceRegisterDialog'] = dialogPolyfill.forceRegisterDialog;
   dialogPolyfill['registerDialog'] = dialogPolyfill.registerDialog;
+
+  if ( typeof module === 'object' && typeof module.exports === 'object' ) {
```
Comment (samthor) [inline dialog-polyfill.js]: So I'm concerned compiling this with Closure (which is optional, but works well) might convert `module` to a tiny variable name. But I'm happy to fix that later.

Comment (Rowno) [inline dialog-polyfill.js]: I don't know the ins and outs of Closure (I've only ever used Uglify), but I figured that people using Closure probably aren't using CommonJS anyway. :wink:

**Comment by samthor** [comment]
@Rowno I've updated it [a touch](https://github.com/GoogleChrome/dialog-polyfill/commit/5b5c20a1115b5f7d1d512a76581a570bd7db8d9e), but this should still work for you.

I'll do a new release soon.

**Comment by Rowno** [comment]
:+1:

**Pull request merged**